spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/EnvironmentControllerTests.java

```diff
@@ -98,6 +99,28 @@ public void arrayInYaml() throws Exception {
 	}
 
 	@Test
+	public void arrayOverridenInYaml() throws Exception {
+		// Add overriden values first, will get reversed when properties are combined
+		Map<String, Object> twoMap = new LinkedHashMap<String, Object>();
+		twoMap.put("a.b[0]", "f");
+		twoMap.put("a.b[1]", "h");
+		this.environment.add(new PropertySource("two", twoMap));
+
+		// Add more values in array than present in first source
+		Map<String, Object> oneMap = new LinkedHashMap<String, Object>();
+		oneMap.put("a.b[0]", "c");
+		oneMap.put("a.b[1]", "d");
+		oneMap.put("a.b[2]", "z");
+		this.environment.add(new PropertySource("one", oneMap));
+
+		Mockito.when(this.repository.findOne("foo", "bar", null)).thenReturn(this.environment);
+		String yaml = this.controller.yaml("foo", "bar", false).getBody();
+
+		// Result will not contain original, extra values from oneMap
+		assertEquals("a:\n  b:\n  - f\n  - h\n", yaml);
```

Comment:
@dsyer Should I add a test similar to others in there like the following:

	@Test
	public void mappingForYaml() throws Exception {
		Mockito.when(this.repository.findOne("foo", "bar", null)).thenReturn(this.environment);
		MockMvc mvc = MockMvcBuilders.standaloneSetup(this.controller).build();
		mvc.perform(MockMvcRequestBuilders.get("/foo-bar.yml"))
		.andExpect(
				MockMvcResultMatchers.content().contentType(MediaType.TEXT_PLAIN))
				.andExpect(MockMvcResultMatchers.content().string("{}\n"));
	}

Is that what you are referring to? If so, I can add another test in there for that.